xds/src/main/java/io/grpc/xds/XdsEndpointResource.java

```diff
@@ -190,22 +193,28 @@ static StructOrError<LocalityLbEndpoints> parseLocalityLbEndpoints(
       if (!endpoint.hasEndpoint() || !endpoint.getEndpoint().hasAddress()) {
         return StructOrError.fromError("LbEndpoint with no endpoint/address");
       }
-      io.envoyproxy.envoy.config.core.v3.SocketAddress socketAddress =
-          endpoint.getEndpoint().getAddress().getSocketAddress();
-      InetSocketAddress addr =
-          new InetSocketAddress(socketAddress.getAddress(), socketAddress.getPortValue());
-      boolean isHealthy =
-          endpoint.getHealthStatus() == io.envoyproxy.envoy.config.core.v3.HealthStatus.HEALTHY
-              || endpoint.getHealthStatus()
-              == io.envoyproxy.envoy.config.core.v3.HealthStatus.UNKNOWN;
+      List<java.net.SocketAddress> addresses = new ArrayList<>();
+      addresses.add(getInetSocketAddress(endpoint.getEndpoint().getAddress()));
+      for (Endpoint.AdditionalAddress additionalAddress
+          : endpoint.getEndpoint().getAdditionalAddressesList()) {
+        addresses.add(getInetSocketAddress(additionalAddress.getAddress()));
+      }
+      boolean isHealthy = (endpoint.getHealthStatus() == HealthStatus.HEALTHY)
+              || (endpoint.getHealthStatus() == HealthStatus.UNKNOWN);
       endpoints.add(Endpoints.LbEndpoint.create(
-          new EquivalentAddressGroup(ImmutableList.<java.net.SocketAddress>of(addr)),
+          new EquivalentAddressGroup(addresses),
```

Comment:
ImmutableList.copyOf()?

Reply:
Addresses is already a new list (created on line 196), so we don't care what is done with it.  On top of that the EquivalentAddressGroup constructor copies whatever we give it.
`this.addrs = Collections.unmodifiableList(new ArrayList<>(addrs));`